source/common/router/scoped_config_impl.cc

```diff
@@ -104,12 +114,36 @@ ScopeKeyBuilderImpl::computeScopeKey(const Http::HeaderMap& headers) const {
   return std::make_unique<ScopeKey>(std::move(key));
 }
 
-void ScopedConfigImpl::addOrUpdateRoutingScope(const ScopedRouteInfoConstSharedPtr&) {}
+void ScopedConfigImpl::addOrUpdateRoutingScope(
+    const ScopedRouteInfoConstSharedPtr& scoped_route_info) {
+  const auto iter = scoped_route_info_by_name_.find(scoped_route_info->scopeName());
+  if (iter != scoped_route_info_by_name_.end()) {
+    scoped_route_info_by_key_.erase(iter->second->scopeKey().hash());
```

Comment:
Would be good to have some ASSERTs around the place to validate consistency of these two data structures.

Reply:
done.